tests/lava/magma/runtime/test_ref_var_ports.py

```diff
@@ -0,0 +1,239 @@
+        sender.stop()
+
+    def test_explicit_Ref_Var_port_read(self):
+        """Tests the connection of a RefPort to a explicitly created VarPort.
+        The RefPort "ref_read" reads data after the first time step of the
+        VarPort "var_port_read" which has the value of the Var "v" (= 17) and
+        writes this value into the Var "var_read". The initial value of the var
+        "var_read" is 1. At time step 2 the value of "var_read" is 17."""
+
+        sender = P1()
+        recv = P2()
+
+        # Connect RefPort with explicit VarPort
+        recv.ref_read.connect(sender.var_port_read)
+
+        class MyRunCfg(RunConfig):
+            def select(self, proc, proc_models):
+                return proc_models[0]
+
+        simple_sync_domain = SyncDomain("simple", LoihiProtocol(),
+                                        [sender, recv])
+
+        # first time step, no read
+        sender.run(RunSteps(num_steps=1, blocking=True),
+                   MyRunCfg(custom_sync_domains=[simple_sync_domain]))
+        # initial value (1) is expected
+        self.assertTrue(np.all(recv.var_read.get() == np.array([1., 1.])))
+        # second time step, the RefPort read from the VarPort and wrote the
+        # result in "var_read" (= 17)
+        sender.run(RunSteps(num_steps=1, blocking=True),
+                   MyRunCfg(custom_sync_domains=[simple_sync_domain]))
+        self.assertTrue(
+            np.all(recv.var_read.get() == np.array([17., 17.])))
+        sender.stop()
+
+    def test_implicit_Ref_Var_port_read(self):
+        """Tests the connection of a RefPort to a implicitly created VarPort.
+        The RefPort "ref_read" reads data after the first time step of the
+        of the Var "v" (= 17) using an implicit VarPort and writes this value
+        into the Var "var_read". The initial value of the var "var_read" is 1.
+        At time step 2 the value of "var_read" is 17."""
+
+        sender = P1()
+        recv = P2()
+
+        # Connect RefPort with explicit VarPort
+        recv.ref_read.connect_var(sender.v)
+
+        class MyRunCfg(RunConfig):
+            def select(self, proc, proc_models):
+                return proc_models[0]
+
+        simple_sync_domain = SyncDomain("simple", LoihiProtocol(),
+                                        [sender, recv])
+
+        # first time step, no read
+        recv.run(RunSteps(num_steps=1, blocking=True),
+                 MyRunCfg(custom_sync_domains=[simple_sync_domain]))
+        # initial value (1) is expected
+        self.assertTrue(np.all(recv.var_read.get() == np.array([1., 1.])))
+        # second time step, the RefPort read from the VarPort and wrote the
+        # result in "var_read" (= 17)
+        recv.run(RunSteps(num_steps=1, blocking=True),
+                 MyRunCfg(custom_sync_domains=[simple_sync_domain]))
+        self.assertTrue(
+            np.all(recv.var_read.get() == np.array([17., 17.])))
+        recv.stop()
```

Comment:
Can we get rid of some of the code-duplication in the unit tests?

Reply:
Done